cpp/include/cudf/column/column_factories.hpp

```diff
@@ -42,13 +44,45 @@ namespace cudf
  * kernels
  * @param[in] mr Optional resource to use for device memory
  * allocation of the column's `data` and `null_mask`.
- *---------------------------------------------------------------------------**/
+ */
 std::unique_ptr<column> make_numeric_column(
     data_type type, size_type size, mask_state state = UNALLOCATED,
     cudaStream_t stream = 0,
     rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
-/**---------------------------------------------------------------------------*
+/**
+ * @brief Construct column with sufficient uninitialized storage
+ * to hold `size` elements of the specified numeric `data_type` with a
+ * null mask.
+ *
+ * @note null_count is optional and will be computed if not provided.
+ *
+ * @throws std::bad_alloc if device memory allocation fails
+ * @throws cudf::logic_error if `type` is not a numeric type
+ *
+ * @param[in] type The desired numeric element type
+ * @param[in] size The number of elements in the column
+ * @param[in] null_mask Null mask to use for this column.
+ * @param[in] null_count Optional number of nulls in the null_mask.
+ * @param[in] stream Optional stream on which to issue all memory allocation and device
+ * kernels
+ * @param[in] mr Optional resource to use for device memory
+ * allocation of the column's `data` and `null_mask`.
+ */
+template <typename B>
+std::unique_ptr<column> make_numeric_column(
+    data_type type, size_type size, 
+    B&& null_mask,
+    size_type null_count = cudf::UNKNOWN_NULL_COUNT, cudaStream_t stream = 0,
+    rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource())
+{
+  CUDF_EXPECTS(is_numeric(type), "Invalid, non-numeric type.");
+  return std::make_unique<column>(
+      type, size, rmm::device_buffer{size * cudf::size_of(type), stream, mr},
+      std::forward(null_mask), null_count);
```

Comment:
Template argument deduction is disabled with `std::forward`
```suggestion
      std::forward<B>(null_mask), null_count);
```